system/lib/standalone_wasm.c

```diff
@@ -51,11 +51,11 @@ void __unlock(void* ptr) {}
 
 void *emscripten_memcpy_big(void *restrict dest, const void *restrict src, size_t n) {
   // This normally calls out into JS which can do a single fast operation,
-  // but with wasi we can't do that. As this is called when n >= 8192, we
+  // but with wasi we can't do that. As this is called when n >= 512, we
   // can just split into smaller calls.
   // TODO optimize, maybe build our memcpy with a wasi variant, maybe have
   //      a SIMD variant, etc.
-  const int CHUNK = 4096;
+  const int CHUNK = 508;
```

Comment:
I wonder if 256 might not be better, as we would have more calls, but they would be perhaps more aligned, which might matter with SIMD eventually? At least worth adding a TODO to investigate for SIMD I think.

Reply:
Alignment doesn't matter much here, since this chunking scheme will kill all performance anyways. The best resolution here will definitely be to remove the whole chunking (like the existing TODO already states) by adjusting to do two lib builds.

Reply:
Where does 508 come from here?

Reply:
It is 512-4, 512 triggers emscripten_memcpy_big, 508 is to keep under that limit while still being 4 byte aligned